app/rtc_compare.py

```diff
@@ -0,0 +1,126 @@
+'''
+Compare the two RTC products if they are equivalent
+Part of the codes are copied from PROTEUS SAS
+'''
+
+import argparse
+
+import h5py
+import numpy as np
+
+RTC_S1_PRODUCTS_ERROR_TOLERANCE = 1e-6
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description='Compare two DSWx-HLS products',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Inputs
+    parser.add_argument('input_file',
+                        type=str,
+                        nargs=2,
+                        help='Input data in HDF5')
+
+    return parser
+
+def compare_hdf5(hdf_obj_1, hdf_obj_2, key_in='/', compare_attr=True):
+    '''
+    Recursively traverse the Dataset and Attributes in the 1st HDF5 (the reference).
+    Try to find the same dataset in the 2nd HDF5 (the target).
+    Returns 0 if the reference and the target HDF5 are identical within threshold
+    Prints out the difference report when they are not.
+
+    Paremeters:
+    ----------
+    hdf_obj_1: Reference HDF5 object
+    hdf_obj_2: Target HDF5 object
+
+    Return:
+    -------
+    _ : 0 if the two HDF are identical; 1 otherwise
+
+    '''
+
+    if isinstance(hdf_obj_1[key_in], h5py.Group):
+        for key_1, _ in hdf_obj_1[key_in].items():
+            if key_1 in hdf_obj_2[key_in].keys():
+                compare_hdf5(hdf_obj_1, hdf_obj_2, f'{key_in}/{key_1}')
```

Comment:
If you keep `compare_attr`, I guess you need to pass it here to the recursive function.

Reply:
the flag `compare_attr` has taken out.